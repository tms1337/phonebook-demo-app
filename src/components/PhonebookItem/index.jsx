import React from "react";
import { colors } from "../../ui";

const PhonebookItem = ({ item: { name, address, phone } }) => {
  return (
    <div style={styles.root}>
      <div style={styles.title}>{name}</div>
      <div style={styles.title}>{address}</div>
      <div style={styles.content}>{phone}</div>
    </div>
  );
};

const styles = {
  root: {
    width: "750px",
    textAlign: "center",
    border: "0.5px solid grey",
    marginTop: "10px",
    marginBottom: "10px",
    padding: "25px"
  },
  title: {
    color: colors.SECONDARY
  },
  content: {
    color: colors.PRIMARY,
    fontWeight: 1000,
    fontSize: "25px"
  }
};

export default PhonebookItem;
