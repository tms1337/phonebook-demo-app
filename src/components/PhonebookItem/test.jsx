import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import PhonebookItem from ".";

describe("PhonebookItem", () => {
  it("renders snapshot", () => {
    const item = {
      name: "Heso",
      surname: "Meso",
      phone: "+007862345"
    };

    const component = renderer.create(<PhonebookItem item={item} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders fields correctly", () => {
    const item = {
      name: "Heso",
      address: "Meso",
      phone: "+007862345"
    };

    const component = shallow(<PhonebookItem item={item} />);

    expect(component.contains(item.name)).toBe(true);
    expect(component.contains(item.address)).toBe(true);
    expect(component.contains(item.phone)).toBe(true);
  });
});
