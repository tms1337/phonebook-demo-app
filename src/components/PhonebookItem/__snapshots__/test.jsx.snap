// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhonebookItem renders snapshot 1`] = `
<div
  style={
    Object {
      "border": "0.5px solid grey",
      "marginBottom": "10px",
      "marginTop": "10px",
      "padding": "25px",
      "textAlign": "center",
      "width": "750px",
    }
  }
>
  <div
    style={
      Object {
        "color": "#9aaeed",
      }
    }
  >
    Heso
  </div>
  <div
    style={
      Object {
        "color": "#9aaeed",
      }
    }
  />
  <div
    style={
      Object {
        "color": "#349beb",
        "fontSize": "25px",
        "fontWeight": 1000,
      }
    }
  >
    +007862345
  </div>
</div>
`;
